lib/elasticity/index_mapper.rb

```diff
@@ -152,8 +152,20 @@ def map_hit(hit)
 
         highlighted = @document_klass.new(highlighted_attrs)
       end
+      if @document_klass.config.subclasses.present?
+        Object.const_get(subclass_types[hit["_type"]]).new(attrs.merge(highlighted: highlighted))
```

Comment:
Rather than doing this here, can we store the "document_type" => "mapping" in a hash that we can use? It will make the code:

* more secure, because you'll be limited to loading objects whitelisted as valid subclasses and document types (it should raise an error otherwise)
* avoid calling const_get for each hit that needs to be instantiated, reflection is expensive